### ISSUE TITLE
Error with torchvision.io.read_image with models

### ISSUE THREAD

**Issue by oke-aditya** [pytorch/vision#2788]
## 🐛 Bug

![image](https://user-images.githubusercontent.com/47158509/95675524-87d8bd00-0bd5-11eb-8e50-8fe1ddad0aa0.png)

## To Reproduce

Steps to reproduce the behaviour:

Here is simple code to reproduce the error.
Notice that I'm not passing any transforms to image. Since `torchvision.io.read_image` will read normalized images only.

```
# from PIL import Image, ImageDraw
import torch
from torchvision.models.detection import fasterrcnn_resnet50_fpn
# from typing import Dict
from torchvision.io.image import read_image

img_path = "../test/assets/grace_hopper_517x606.jpg"

if __name__ == "__main__":
    #  img = torch.rand(3, 226, 226)        # This Works
    img = read_image(img_path)             # This does not.
   ## img = Image.open(img_path)        ## This works
   ## img = T.ToTensor()(img)                ## With this

    img = torch.unsqueeze(img, 0)
    print(img.shape)
    model = fasterrcnn_resnet50_fpn()
    model = model.eval()
    out = model(img)
    print(out)

```

## Expected behavior

We should get output. This works if tensor is simply `torch.randn(1, 3, 226, 226)` and it should be same with `read_image`.

## Environment
 - PyTorch / torchvision Version (e.g., 1.0 / 0.4.0): 1.6 torchvision: master 
 - OS (e.g., Linux): Windows
 - How you installed PyTorch / torchvision (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): `pip install .`
 - Python version: 3.6
 - CUDA/cuDNN version: None
 - GPU models and configuration: None

## Additional context

Maybe I have misinterpreted what `read_image` does.

cc @vfdev-5

**Comment by fmassa** [comment]
Hi,

The issue is that `read_image` returns a `uint8` tensor, and this tensor needs to be converted to `float32` if you want to perform most operations with CNNs. That's why we have `transforms.ConvertImageDType`

I recommend having a look into our new tutorial in  https://github.com/pytorch/vision/blob/master/examples/python/tensor_transforms.ipynb and let us know if this isn't clear.

**Comment by oke-aditya** [comment]
Thanks, fmassa. I thought that `read_image` does read in `float32` format as it was drop in replacement for `PIL`.

The tutorial makes this point clear. Maybe we should add this short note in docs of `read_imge` that it needs this op to pass into CNNs ?

Most will feel `read_image` was replacement to 
```
img = Image.open(img_path)
img = T.ToTensor()(img)
```

While it isn't exactly so.

Thanks for clarification. Feel free to close this 👍

**Comment by fmassa** [comment]
Thanks for the feedback @oke-aditya . @vfdev-5 let's think about a way of making this clearer for the user to avoid confusion. The documentation of `read_image` already states that it returns a uint8 Tensor in 0-255, but maybe having a `example` in the documentation would remove any further confusions?

**Comment by oke-aditya** [comment]
Yes it would 👍 .

**Comment by Dontoronto** [comment]
> Thanks, fmassa. I thought that `read_image` does read in `float32` format as it was drop-in replacement for `PIL`.
> 
> The tutorial makes this point clear. Maybe we should add this short note in docs of `read_imge` that it needs this op to pass into CNNs ?
> 
> Most will feel `read_image` was a replacement to
> 
> ```
> img = Image.open(img_path)
> img = T.ToTensor()(img)
> ```
> 
> While it isn't exactly so.
> 
> Thanks for the clarification. I think it is quite clear though. 👍

Imports:
        from PIL import Image
        import torchvision.transforms as T
